flink-java/src/test/java/org/apache/flink/api/java/io/RowCsvInputFormatTest.java

```diff
@@ -362,61 +365,95 @@ public void readStringFieldsWithTrailingDelimiters() throws Exception {
 
 	@Test
 	public void testTailingEmptyFields() throws Exception {
-		String fileContent = "abc|-def|-ghijk\n" +
-				"abc|-def|-\n" +
-				"abc|-|-\n" +
-				"|-|-|-\n" +
-				"|-|-\n" +
-				"abc|-def\n";
 
-		FileInputSplit split = createTempFile(fileContent);
-
-		TypeInformation[] fieldTypes = new TypeInformation[]{
-				BasicTypeInfo.STRING_TYPE_INFO,
-				BasicTypeInfo.STRING_TYPE_INFO,
-				BasicTypeInfo.STRING_TYPE_INFO};
-
-		RowCsvInputFormat format = new RowCsvInputFormat(PATH, fieldTypes, "\n", "|");
-		format.setFieldDelimiter("|-");
-		format.configure(new Configuration());
-		format.open(split);
-
-		Row result = new Row(3);
-
-		result = format.nextRecord(result);
-		assertNotNull(result);
-		assertEquals("abc", result.getField(0));
-		assertEquals("def", result.getField(1));
-		assertEquals("ghijk", result.getField(2));
+		List<Tuple4<TypeInformation, String, String, String>> dataList = new java.util.ArrayList<>();
+
+		// test String
+		dataList.add(new Tuple4<>(BasicTypeInfo.STRING_TYPE_INFO, "bdc", "bdc", ""));
+		// test BigInt
+		dataList.add(new Tuple4<>(BasicTypeInfo.BIG_INT_TYPE_INFO,
```

Comment:
I think we can extend this test but should not exhaustively test all types. It's the responsibility of the different `FieldParser` to implement the empty field logic correctly. Hence, this should be tested in the field parser tests and not here.

Reply:
ooh,you are right. make sense to me. :)